### PR TITLE
web: fix update button callback

### PR DESCRIPTION
**Pull request by maxime-carabina**
We are updating callback for button update in `Settings` page.

### CODE REVIEW

File: web-frontend/src/mirage/server.js [hunk 1]
```diff
@@ -47,9 +47,14 @@ function mockServer(environment = ENV.DEV) {
       });
 
       this.put('accounts/:nickname', (schema, request) => {
-        let { newNickname } = JSON.parse(request.requestBody).newNickname;
+        let { newNickname } = JSON.parse(request.requestBody);
+        let nickname = request.params.nickname;
+
+        console.log('mirage');
```
Comment (Thykof) [inline web-frontend/src/mirage/server.js:53]:
```suggestion
```

File: web-frontend/src/mirage/server.js
Comment (Thykof) [inline web-frontend/src/mirage/server.js:1]: from typescript to javascript... I don't like it that much

File: web-frontend/tsconfig.json [hunk 2]
```diff
@@ -5,15 +5,14 @@
     "module": "ESNext",
     "skipLibCheck": true,
 
-    /* Bundler mode */
     "moduleResolution": "bundler",
     "allowImportingTsExtensions": true,
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
+    "allowJs": true,
```
Comment (Thykof) [inline web-frontend/tsconfig.json:13]:
```suggestion
```

File: web-frontend/src/pages/Settings/SeetingsEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            </Button>
+            <Button
+              preIcon={<FiTrash2 />}
+              onClick={() => mutableDelete({} as AccountObject)}
+              variant="danger"
+            >
+              {Intl.t('settings.buttons.delete')}
+            </Button>
+          </div>
+        </div>
```
Comment (Thykof) [inline web-frontend/src/pages/Settings/SeetingsEdit.tsx:84]: code duplication with `web-frontend/src/pages/Settings/Settings.tsx`

File: web-frontend/src/main.tsx [hunk 4]
```diff
@@ -50,8 +51,9 @@ const router = createBrowserRouter(
       <Route path=":nickname/send-coins" element={<SendCoins />} />
       <Route path=":nickname/receive" element={<ReceiveCoins />} />
       <Route path=":nickname/assets" element={<></>} />
-      <Route path=":nickname/settings" element={<Settings />} />
       <Route path=":nickname/contacts" element={<></>} />
+      <Route path=":nickname/settings" element={<Settings />} />
+      <Route path=":nickname/settings/update" element={<SettingsEdit />} />
```
Comment (Thykof) [inline web-frontend/src/main.tsx:56]: why do we need a new page?

**Review by Thykof — CHANGES_REQUESTED** [review]

File: web-frontend/src/main.tsx (hunk 4 above)
Comment (mazmassa) [reply web-frontend/src/main.tsx:56]: We need to change the route to update the nickname in the URL (at least for now).
The page we are adding has the edit button that redirects when is success.

BTW, It's pretty fluid and match with the first design.
We can check that together if you prefer!

File: web-frontend/src/pages/Settings/SeetingsEdit.tsx (hunk 3 above)
Comment (mazmassa) [reply web-frontend/src/pages/Settings/SeetingsEdit.tsx:84]: True! We could extract that.

Comment (maxime-carabina) [reply web-frontend/src/pages/Settings/SeetingsEdit.tsx:84]: > code duplication with `web-frontend/src/pages/Settings/Settings.tsx`

Exactly! I will put this part in another file name `SettingsOption`.

Comment (maxime-carabina) [reply web-frontend/src/pages/Settings/SeetingsEdit.tsx]: Done

File: web-frontend/src/mirage/server.js (hunk 1 above)
Comment (maxime-carabina) [reply web-frontend/src/mirage/server.js]: Done

File: web-frontend/src/mirage/server.js
Comment (Thykof) [reply web-frontend/src/mirage/server.js:1]: created issue #338

**Review by Thykof — APPROVED** [review]

**Pull request merged**